src/genericSigning.ts

```diff
@@ -232,21 +232,23 @@ export const parseGenericSigningResponse = function (res, off, req) {
       off += 65;
     }
     // Handle `GpECDSASig_t`
-    parsed.sig = parseDER(res.slice(off, off + 2 + res[off + 1]));
+    const derSig = parseDER(res.slice(off, off + 2 + res[off + 1]));
     // Remove any leading zeros in signature components to ensure
     // the result is a 64 byte sig
-    parsed.sig.r = fixLen(parsed.sig.r, 32);
-    parsed.sig.s = fixLen(parsed.sig.s, 32);
+    const rBuf = fixLen(derSig.r, 32);
+    const sBuf = fixLen(derSig.s, 32);
 
-    // If this is an EVM request, we want to add a `v` and format r,s as hex strings with 0x prefix
-    if (req.encodingType === Constants.SIGNING.ENCODINGS.EVM) {
-      const vBn = getV(req.origPayloadBuf, parsed);
-      // Convert v to hex string for consistency with r and s
-      parsed.sig.v = `0x${vBn.toString(16)}`;
+    parsed.sig = {
+      r: `0x${rBuf.toString('hex')}`,
+      s: `0x${sBuf.toString('hex')}`,
+    };
 
-      // Format r and s as hex strings with 0x prefix for consistency with legacy ETH signing
-      parsed.sig.r = `0x${parsed.sig.r.toString('hex')}`;
-      parsed.sig.s = `0x${parsed.sig.s.toString('hex')}`;
+    if (
+      req.encodingType === Constants.SIGNING.ENCODINGS.EVM ||
+      req.hashType === Constants.SIGNING.HASHES.KECCAK256
+    ) {
+      const vBn = getV(req.origPayloadBuf, parsed);
+      parsed.sig.v = BigInt(vBn.toString());
```

Comment:
I see the `viem/Signature` implementation and also the legacy eth transactions have the type of `v` as `bigint` :white_check_mark:
Should we make it optional and update other places to `bigint` like https://github.com/GridPlus/gridplus-sdk/blob/6790828e26b8a9425a5a915c7329b64831d80497/docs/docs/signing.md?plain=1#L203 ?

And also where do we have the following definition, that the Kalo mentioned in his [issue](https://github.com/GridPlus/gridplus-sdk/issues/611)?

```interface SignData {
  sig?: {
    v: Buffer;
    r: Buffer;
    s: Buffer;
  };
  ...
}
```